src/hooks/use-cart.ts

```diff
@@ -22,28 +25,54 @@ export const useCart = () => {
 
     const updateItemQuantity = ({ id, quantity }: { id: string; quantity: number }) => {
         setUpdatingCartItems((prev) => [...prev, id]);
-        triggerUpdateItemQuantity({ id, quantity }).finally(() => {
-            setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id));
-        });
+        triggerUpdateItemQuantity({ id, quantity })
+            .catch((error) => {
+                runToast({
+                    type: ToastType.Error,
+                    message: `Failed to update the item quantity. ${getErrorMessage(error)}`,
+                });
+                throw error;
+            })
+            .finally(() => {
+                setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id));
+            });
     };
 
     const removeItem = (id: string) => {
         setUpdatingCartItems((prev) => [...prev, id]);
-        triggerRemoveItem(id).finally(() => {
-            setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id));
-        });
+        triggerRemoveItem(id)
+            .catch((error) => {
+                runToast({
+                    type: ToastType.Error,
+                    message: `Failed to remove the item. ${getErrorMessage(error)}`,
+                });
+                throw error;
+            })
+            .finally(() => {
+                setUpdatingCartItems((prev) => prev.filter((itemId) => itemId !== id));
+            });
     };
 
     const addToCart = (productId: string, quantity: number, options?: AddToCartOptions) =>
-        triggerAddToCart({ id: productId, quantity, options });
+        triggerAddToCart({ id: productId, quantity, options }).catch((error) => {
+            runToast({
+                type: ToastType.Error,
+                message: `Failed to add product to the cart. ${getErrorMessage(error)}`,
+            });
+            throw error;
+        });
 
     const checkout = async () => {
         const checkoutResponse = await ecomAPI.checkout();
 
         if (checkoutResponse.status === 'success') {
             window.location.href = checkoutResponse.body.checkoutUrl;
         } else {
-            alert('checkout is not configured');
+            runToast({
+                type: ToastType.Error,
```

Comment:
1. Just pass strings for type, does TypeScript allow using strings here?

2. I'm not sure it's a good idea to put toasts into the useCart hook, rather than calling `checkout().catch(() => toast.error('Oops'))`. If the template author wants to show the error in the form. @Letushev what do you think?

Reply:
I think it shouldn't be part of `useCart` hook.
>  If the template author wants to show the error in the form

True. Also other templates might want to use different toast message or duration.